### PR TITLE
feat: Add support for BitBucket Cloud (#311)

### PR DESCRIPTION
**Pull request by SergK**
This pull request introduces enhancements to the `epam/edp-tekton` component to support BitBucket Cloud. The modifications include updates to the Go codebase and adjustments to Tekton Pipelines and TriggerTemplates to ensure compatibility and functionality with BitBucket Cloud events. This integration is crucial for improving our project's operational flexibility and scalability with BitBucket Cloud.

Related issue #311 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## How Has This Been Tested?
Testing was conducted in a staged environment with the following steps:
1. Deployed the updated `edp-tekton` component with the new BitBucket Cloud support features.
2. Created a mock BitBucket Cloud event to trigger the updated pipelines.
3. Verified that the pipelines executed as expected and interfaced correctly with BitBucket Cloud.

Further integration testing and end-to-end testing with real BitBucket Cloud events are planned to ensure robust functionality.

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Pull Request contains one commit. I squash my commits.

## Screenshots (if appropriate):
(If applicable, add screenshots to help explain the changes)

## Additional context
This pull request is part of a larger effort to enhance the `edp-tekton` component's compatibility and functionality with various version control and CI/CD tools, specifically focusing on BitBucket Cloud in this instance. The successful integration of these features will significantly benefit our project's efficiency and interoperability with BitBucket Cloud.

### CODE REVIEW

File: pkg/event_processor/bitbucket/bitbucket.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		TargetBranch:       bitbucketEvent.PullRequest.Destination.Branch.Name,
+		Type:               event_processor.EventTypeReviewComment,
+		Codebase:           codebase,
+		HasPipelineRecheck: event_processor.ContainsPipelineRecheck(bitbucketEvent.Comment.Content.Raw),
+		PullRequest: &event_processor.PullRequest{
+			HeadSha:           bitbucketEvent.PullRequest.Source.Commit.Hash,
+			Title:             bitbucketEvent.PullRequest.Title,
+			HeadRef:           bitbucketEvent.PullRequest.Source.Branch.Name,
+			ChangeNumber:      bitbucketEvent.PullRequest.ID,
+			LastCommitMessage: bitbucketEvent.PullRequest.LastCommit.Hash,
```
Comment (zmotso) [inline pkg/event_processor/bitbucket/bitbucket.go:102]: LastCommit hash, not message?

File: pkg/event_processor/event.go [hunk 2]
```diff
@@ -52,14 +52,69 @@ type GitLabComment struct {
...
 }
 
-const (
-	GitProviderGitHub = "github"
-	GitProviderGitLab = "gitlab"
-	GitProviderGerrit = "gerrit"
+type BitbucketEvent struct {
+	Repository  BitbucketRepository  `json:"repository"`
+	PullRequest BitbucketPullRequest `json:"pullrequest"`
+	Actor       BitbucketActor       `json:"actor"`
```
Comment (zmotso) [inline pkg/event_processor/event.go:58]: Do we need Actor ?

File: pkg/event_processor/bitbucket/bitbucket.go (hunk 1 above)
Comment (SergK) [reply pkg/event_processor/bitbucket/bitbucket.go:102]: Discussed, we'll address it in the next patch

**Pull request merged**